### ISSUE TITLE
Errors on redirect

### ISSUE THREAD

**Issue by mvader** [lotus/controller#99]
Hi,

I've been trying to get the errors of the previous action after a redirect (which I've seen it's possible according to [lotus/controller code](https://github.com/lotus/controller/blob/master/lib/lotus/action/session.rb#L114)) but I can't. I've also been searching for examples on how to do this right but I found nothing.

I have this on my controller config:
```ruby
cookies true
sessions :cookie, secret: "foo"
```
On my action after the redirect I have this:
```ruby
def call(params)
  puts errors.inspect
end
```

The first time I submit my form everything is ok, I get the errors on the action after the redirect. But if I submit the form more times I get the same errors as the first time no matter what the input contents are.

I've also noticed that two cookies "rack.session" are set. One with the path "/" and one with the path "/foo" where /foo is the endpoint of the form action. Maybe it has something to do with that.

**Comment by jodosha** [comment]
Hello @mvader Thanks for opening this ticket. Are you using this framework alone, or with `lotusrb`?

**Comment by mvader** [comment]
I'm using it with lotusrb. 
On Apr 7, 2015 11:08 AM, Luca Guidi <notifications@github.com> wrote:Hello @mvader Thanks for opening this ticket. Are you using this framework alone, or with lotusrb?

**Comment by jodosha** [comment]
@mvader Which kind of error do you get? Is this related to https://github.com/lotus/lotus/issues/196 ?

**Comment by mvader** [comment]
I don't get any error. It just doesn't work as expected. I've been able to reproduce even with the fix you mentioned in #196. 

Just setup an application with cookies and the session like that:
```ruby
cookies true
sessions :cookie, secret: ENV['WEB_SESSIONS_SECRET']
```

Once you've submitted a form with errors, on every other request the retrieved errors via the ```errors``` method are the same, no matter what the input parameters were.

As I said, two cookies are set, maybe that's the problem. I implemented the sessions and cookies myself using Rack::Session and it worked. With Rack::Session only one cookie was set.
